src/Workspaces/CSharp/Portable/Classification/SyntaxClassification/NameSyntaxClassifier.cs

```diff
@@ -182,6 +182,31 @@ CandidateReason.Ambiguous or
                 return true;
             }
 
+            if (name is IdentifierNameSyntax { Identifier.Text: "async" } && symbol is null)
```

Comment:
>             [NotNullWhen(returnValue: true)] ISymbol? symbol,

:)

Reply:
This was copy-pasted from seveal lines above, I am not responsible for this) I don't know how exactly this attribute works as I don't have much expirience in working with nullability. Can you tell, what is wrong here?

Reply:
Basically the attribute says that if this method returned true, the parameter `symbol` can never be null. This method (`TryClassifySymbol`) is responsible for classifying a symbol. So it shouldn't return true for a null symbol IMO.

Maybe this should be done in `TryClassifyFromIdentifier` or some other "new" method?

I'm not sure what the best option is, so @CyrusNajmabadi can guide you better :)

Reply:
i agree. this should not be in TryClassifySymbol (because there is no symbol).

Reply:
@CyrusNajmabadi Is it in the right spot now?